libherokubuildpack/src/buildpack_output/mod.rs

```diff
@@ -491,6 +524,40 @@ where
         }
     }
 
+    /// Output periodic timer updates to the end user.
+    ///
+    /// If a buildpack author wishes to start a long-running task that does not stream, starting a timer
+    /// will let the user know that the buildpack is performing work and that the UI is not stuck.
+    ///
+    /// One common use case is when downloading a file. Emitting periodic output when downloading is especially important for the local
+    /// buildpack development experience where the user's network may be unexpectedly slow, such as
+    /// in a hotel or on a plane.
+    ///
+    /// This function will transition your buildpack output to [`state::Background`].
+    #[allow(clippy::missing_panics_doc)]
```

Comment:
This is because of manually writing and flushing the IO that appends to the same line. We need to do that here because `writeln_now` writes...lines. I originally had two functions `write_now` and `writeln_now` and it was a tad hard to read and know which case to expect.